deployment/common/changeset/link_transfer_timelock.go

```diff
@@ -0,0 +1,177 @@
+package changeset
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	owner_helpers "github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+)
+
+type LinkTransfer struct {
+	To    common.Address
+	Value *big.Int
+}
+type LinkTransferTimelockConfig struct {
+	Transfers  map[uint64][]LinkTransfer
+	UseMCMS    bool
+	McmsConfig *MCMSConfig
+}
+type MCMSConfig struct {
+	ValidUntil   uint32        // unix time until the proposal will be valid
+	MinDelay     time.Duration // delay for timelock worker to execute the transfers.
+	OverrideRoot bool
+}
+
+var _ deployment.ChangeSet[*LinkTransferTimelockConfig] = LinkTransferTimelock
+
+// Validate checks that the LinkTransferTimelockConfig is valid.
+func (cfg LinkTransferTimelockConfig) Validate() error {
+
+	// Check that Transfers map has at least one key
+	if len(cfg.Transfers) == 0 {
+		return fmt.Errorf("transfers map must have at least one key")
+	}
+
+	// Check transfers config values.
+	for key, transfers := range cfg.Transfers {
+		if len(transfers) == 0 {
+			return fmt.Errorf("transfers for key %d must have at least one LinkTransfer", key)
+		}
+		for _, transfer := range transfers {
+			if transfer.To == (common.Address{}) {
+				return fmt.Errorf("'to' address for transfers  must be set")
+			}
+			if transfer.Value == nil {
+				return fmt.Errorf("value for transfers must be set")
+			}
+			if transfer.Value == big.NewInt(0) {
+				return fmt.Errorf("value for transfers must be non-zero")
+			}
+		}
+	}
+	// Check that Transfers and StartingOpCount have the same keys
+	for key := range cfg.Transfers {
+		_, err := chain_selectors.GetSelectorFamily(key)
+		if err != nil {
+			return fmt.Errorf("invalid chain selector: %w", err)
+		}
+	}
+	if !cfg.UseMCMS {
+		return nil
+	}
+	// Mcms specific configs
+	if cfg.McmsConfig == nil {
+		return fmt.Errorf("mcmsConfig must be set when UseMCMS is true")
+	}
+	// Upper bound for min delay (7 days)
+	if cfg.McmsConfig.MinDelay > 24*7*time.Hour {
```

Comment:
can used a named const at top of file like MaxTimelockDelay

Reply:
added https://github.com/smartcontractkit/chainlink/pull/15512/commits/6641f4f9112ee08257fa68e04bcc8d751cc649d8